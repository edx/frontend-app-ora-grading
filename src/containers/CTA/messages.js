/* eslint-disable quotes */
import { defineMessages } from '@edx/frontend-platform/i18n';
import { StrictDict } from 'utils';

const messages = defineMessages({
  ctaFeedbackMessage: {
    id: 'ora-grading.CTA.feedbackMessage',
    defaultMessage: 'Thanks for using the new ORA staff grading experience. We\'d love to hear your feedback',
    description: 'Thank user for using ora and ask for feed back',
  },
  ctaLinkMessage: {
    id: 'ora-grading.CTA.linkMessage',
    defaultMessage: 'here',
    description: 'placeholder for the feedback anchor link',
  },
});

export default StrictDict(messages);
