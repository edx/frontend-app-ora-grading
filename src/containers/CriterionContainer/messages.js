import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  addComments: {
    id: 'ora-grading.CriterionFeedback.addCommentsLabel',
    defaultMessage: 'Add comments',
    description: 'label for editable feedback field',
  },
  comments: {
    id: 'ora-grading.CriterionFeedback.commentsLabel',
    defaultMessage: 'Comments',
    description: 'label for read-only feedback field',
  },
  optionPoints: {
    id: 'ora-grading.RadioCriterion.optionPoints',
    defaultMessage: '{points} points',
    description: 'criterion option point value display',
  },
});

export default messages;
