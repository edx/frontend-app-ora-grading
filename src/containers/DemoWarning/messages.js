/* eslint-disable quotes */
import { defineMessages } from '@edx/frontend-platform/i18n';
import { StrictDict } from 'utils';

const messages = defineMessages({
  demoModeHeading: {
    id: 'ora-grading.ReviewModal.demoHeading',
    defaultMessage: 'Demo Mode',
    description: 'Demo mode heading',
  },
  demoModeMessage: {
    id: 'ora-grading.ReviewModal.demoMessage',
    defaultMessage: 'You are using the Demo Mode of the new Enhanced ORA Staff Grader interface.  You will be unable to submit grades until you activate the feature.',
    description: 'Demo mode message',
  },
});

export default StrictDict(messages);
