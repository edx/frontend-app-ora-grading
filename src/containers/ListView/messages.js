import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  backToResponses: {
    id: 'ora-grading.ListView.ListViewBreadcrumbs.backToResponses',
    defaultMessage: 'Back to all open responses',
    description: 'Breadcrumbs link text to return to ORA list in LMS.',
  },
  noResultsFound: {
    id: 'ora-grading.ListView.noResultsFound',
    defaultMessage: 'No results found',
    description: 'Empty table content for submissions list',
  },
  viewAllResponses: {
    id: 'ora-grading.ListView.viewAllResponses',
    defaultMessage: 'View all responses',
    description: 'Button text to load all responses for review/grading',
  },
  viewSelectedResponses: {
    id: 'ora-grading.ListView.viewSelectedResponses',
    defaultMessage: 'View selected responses ({value})',
    description: 'Button text to load selected responses for review/grading',
  },
  username: {
    id: 'ora-grading.ListView.tableHeaders.username',
    defaultMessage: 'Username',
    description: 'Username table column header for submission list view',
  },
  teamName: {
    id: 'ora-grading.ListView.tableHeaders.teamName',
    defaultMessage: 'Team name',
    description: 'Team name table column header for submission list view',
  },
  learnerSubmissionDate: {
    id: 'ora-grading.ListView.tableHeaders.learnerSubmissionDate',
    defaultMessage: 'Learner submission date',
    description: 'Learner submission date table column header for submission list view',
  },
  teamSubmissionDate: {
    id: 'ora-grading.ListView.tableHeaders.teamSubmissionDate',
    defaultMessage: 'Team submission date',
    description: 'Team submission date table column header for submission list view',
  },
  grade: {
    id: 'ora-grading.ListView.tableHeaders.grade',
    defaultMessage: 'Grade',
    description: 'Grade table column header for submission list view',
  },
  gradingStatus: {
    id: 'ora-grading.ListView.tableHeaders.gradingStatus',
    defaultMessage: 'Grading status',
    description: 'Grading status table column header for submission list view',
  },
});

export default messages;
