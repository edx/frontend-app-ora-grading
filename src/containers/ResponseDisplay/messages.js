import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  tableNameHeader: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.tableNameHeader',
    defaultMessage: 'Name',
    description: 'Table header file name',
  },
  tableExtensionHeader: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.tableExtensionHeader',
    defaultMessage: 'File Extension',
    description: 'Table header file extension',
  },
  tablePopoverHeader: {
    id: 'ora-grading.ResponseDisplay.SubmissionFiles.tablePopoverHeader',
    defaultMessage: 'File Metadata',
    description: 'Table header for popover file metadata',
  },
});

export default messages;
