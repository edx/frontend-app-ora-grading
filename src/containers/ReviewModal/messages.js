import { defineMessages } from '@edx/frontend-platform/i18n';
import { StrictDict } from 'utils';

const messages = defineMessages({
  reloadSubmission: {
    id: 'ora-grading.ReviewModal.reloadSubmission',
    defaultMessage: 'Reload submission',
    description: 'Reload button text in case of network failure',
  },
  loadingResponse: {
    id: 'ora-grading.ReviewModal.loadingResponse',
    defaultMessage: 'Loading response',
    description: 'loading text for submission response review screen',
  },
  gradeNotSubmittedHeading: {
    id: 'ora-grading.ReviewModal.gradeNotSubmitted.heading',
    defaultMessage: 'Grade not submitted',
    description: 'Grade submission network error heading',
  },
  gradeNotSubmittedContent: {
    id: 'ora-grading.ReviewModal.gradeNotSubmitted.heading',
    defaultMessage: "We're sorry, something went wrong when we tried to submit this grade.  Please try again.",
    description: 'Grade submission network error message',
  },
  resubmitGrade: {
    id: 'ora-grading.ReviewModal.resubmitGrade',
    defaultMessage: 'Resubmit grate',
    description: 'Resubmit grade button after network failure',
  },
  dismiss: {
    id: 'ora-grading.ReviewModal.dismiss',
    defaultMessage: 'Dismiss',
    description: 'Dismiss error action button text',
  },
});

export default StrictDict(messages);
