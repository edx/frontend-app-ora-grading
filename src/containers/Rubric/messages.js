import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  rubric: {
    id: 'ora-grading.Rubric.rubric',
    defaultMessage: 'Rubric',
    description: 'Rubric interface label',
  },
  submitGrade: {
    id: 'ora-grading.Rubric.submitGrade',
    defaultMessage: 'Submit grade',
    description: 'Submit Grade button text',
  },
  overallComments: {
    id: 'ora-grading.Rubric.overallComments',
    defaultMessage: 'Overall comments',
    description: 'Rubric overall commnents label',
  },
  addComments: {
    id: 'ora-grading.Rubric.addComments',
    defaultMessage: 'Add comments',
    description: 'Rubric comments input label',
  },
  comments: {
    id: 'ora-grading.Rubric.comments',
    defaultMessage: 'Comments',
    description: 'Rubric comments display label',
  },
});

export default messages;
