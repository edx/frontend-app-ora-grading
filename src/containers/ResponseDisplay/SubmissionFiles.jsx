import React from 'react';
import PropTypes from 'prop-types';

import {
  Card, Collapsible, Icon, DataTable,
} from '@edx/paragon';
import { ArrowDropDown, ArrowDropUp } from '@edx/paragon/icons';
import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';

import FileNameCell from './components/FileNameCell';
import FileExtensionCell from './components/FileExtensionCell';
import FilePopoverCell from './components/FilePopoverCell';

import messages from './messages';

/**
 * <SubmissionFiles />
 */
export class SubmissionFiles extends React.Component {
  get title() {
    return `Submission Files (${this.props.files.length})`;
  }

  render() {
    const { files, intl } = this.props;
    return (
      <Card className="submission-files">
        {files.length ? (
          <Collapsible.Advanced defaultOpen>
            <Collapsible.Trigger className="submission-files-title">
              <h3>{this.title}</h3>
              <Collapsible.Visible whenClosed>
                <Icon src={ArrowDropDown} />
              </Collapsible.Visible>
              <Collapsible.Visible whenOpen>
                <Icon src={ArrowDropUp} />
              </Collapsible.Visible>
            </Collapsible.Trigger>
            <Collapsible.Body className="submission-files-body">
              <DataTable
                columns={[
                  {
                    Header: intl.formatMessage(messages.tableNameHeader),
                    accessor: 'name',
                    Cell: FileNameCell,
                  },
                  {
                    Header: intl.formatMessage(messages.tableExtensionHeader),
                    accessor: 'name',
                    id: 'extension',
                    Cell: FileExtensionCell,
                  },
                  {
                    Header: intl.formatMessage(messages.tablePopoverHeader),
                    accessor: '',
                    Cell: FilePopoverCell,
                  },
                ]}
                data={files}
                itemCount={files.length}
              >
                <DataTable.Table />
              </DataTable>
            </Collapsible.Body>
          </Collapsible.Advanced>
        ) : (
          <div className="submission-files-title no-submissions">
            <h3>{this.title}</h3>
          </div>
        )}
      </Card>
    );
  }
}

SubmissionFiles.defaultProps = {
  files: [],
};
SubmissionFiles.propTypes = {
  files: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      description: PropTypes.string,
      downloadURL: PropTypes.string,
    }),
  ),
  intl: intlShape.isRequired,
};

export default injectIntl(SubmissionFiles);
