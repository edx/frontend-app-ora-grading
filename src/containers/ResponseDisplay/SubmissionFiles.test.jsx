import React from 'react';
import { shallow } from 'enzyme';

import { downloadAllLimit, downloadSingleLimit } from 'data/constants/files';

import { formatMessage } from 'testUtils';
import { SubmissionFiles } from './SubmissionFiles';
import messages from './messages';

jest.mock('./components/FileNameCell', () => jest.fn().mockName('FileNameCell'));
jest.mock('./components/FileExtensionCell', () => jest.fn().mockName('FileExtensionCell'));
jest.mock('./components/FilePopoverCell', () => jest.fn().mockName('FilePopoverCell'));
jest.mock('./FileDownload', () => 'FileDownload');

describe('SubmissionFiles', () => {
  describe('component', () => {
    const props = {
      files: [
        {
          name: 'some file name.jpg',
          description: 'description for the file',
          downloadURL: '/valid-url-wink-wink',
          size: 0,
        },
        {
          name: 'file number 2.jpg',
          description: 'description for this file',
          downloadURL: '/url-2',
          size: 0,
        },
      ],
    };
    let el;
    beforeEach(() => {
      el = shallow(<SubmissionFiles intl={{ formatMessage }} {...props} />);
    });

    describe('snapshot', () => {
      test('files existed for props', () => {
        expect(el).toMatchSnapshot();
      });

      test('files does not exist', () => {
        el.setProps({ files: [] });

        expect(el).toMatchSnapshot();
      });
      test('files size exceed', () => {
        const files = props.files.map(file => ({ ...file, size: downloadSingleLimit + 1 }));
        el.setProps({ files });
        expect(el).toMatchSnapshot();
      });
    });

    describe('behavior', () => {
      test('title', () => {
        const titleEl = el.find('.submission-files-title>h3');
        expect(titleEl.text()).toEqual(
          `${formatMessage(messages.submissionFiles)} (${props.files.length})`,
        );
        expect(el.instance().title).toEqual(
          `${formatMessage(messages.submissionFiles)} (${props.files.length})`,
        );
      });

      describe('canDownload', () => {
        test('normal file size', () => {
          expect(el.instance().canDownload).toEqual(true);
        });

        test('one of the file exceed the limit', () => {
          const oneFileExceed = [{ ...props.files[0], size: downloadSingleLimit + 1 }, props.files[1]];

          oneFileExceed.forEach(file => expect(file.size < downloadAllLimit).toEqual(true));

          el.setProps({ files: oneFileExceed });
          expect(el.instance().canDownload).toEqual(false);

          const warningEl = el.find('span.exceed-download-text');
          expect(warningEl.text().trim()).toEqual(formatMessage(messages.exceedFileSize));
        });

        test('total file size exceed the limit', () => {
          const length = 20;
          const totalFilesExceed = new Array(length).fill({
            name: 'some file name.jpg',
            description: 'description for the file',
            downloadURL: '/valid-url-wink-wink',
            size: (downloadAllLimit + 1) / length,
          });
          totalFilesExceed.forEach(file => {
            expect(file.size < downloadAllLimit).toEqual(true);
            expect(file.size < downloadSingleLimit).toEqual(true);
          });

          el.setProps({ files: totalFilesExceed });
          expect(el.instance().canDownload).toEqual(false);
        });
      });
    });
  });
});
