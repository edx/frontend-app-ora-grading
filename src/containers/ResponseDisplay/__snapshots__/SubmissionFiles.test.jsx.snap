// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionFiles component snapshot files does not exist 1`] = `
<Card
  className="submission-files"
>
  <div
    className="submission-files-title no-submissions"
  >
    <h3>
      Submission Files (0)
    </h3>
  </div>
</Card>
`;

exports[`SubmissionFiles component snapshot files exited for props 1`] = `
<Card
  className="submission-files"
>
  <Collapsible.Advanced
    defaultOpen={true}
  >
    <Collapsible.Trigger
      className="submission-files-title"
    >
      <h3>
        Submission Files (2)
      </h3>
      <Collapsible.Visible
        whenClosed={true}
      >
        <Icon
          src={[MockFunction Icons.ArrowDropDown]}
        />
      </Collapsible.Visible>
      <Collapsible.Visible
        whenOpen={true}
      >
        <Icon
          src={[MockFunction Icons.ArrowDropUp]}
        />
      </Collapsible.Visible>
    </Collapsible.Trigger>
    <Collapsible.Body
      className="submission-files-body"
    >
      <DataTable
        columns={
          Array [
            Object {
              "Cell": [MockFunction HeaderEllipsesCell],
              "Header": "Name",
              "accessor": "name",
            },
            Object {
              "Cell": [MockFunction HeaderFileExtensionCell],
              "Header": "File Extension",
              "accessor": "name",
              "id": "extension",
            },
            Object {
              "Cell": [MockFunction HeaderInfoPopoverCell],
              "Header": "Popover",
              "accessor": "",
            },
          ]
        }
        data={
          Array [
            Object {
              "description": "description for the file",
              "downloadUrl": "/valid-url-wink-wink",
              "name": "some file name.jpg",
            },
            Object {
              "description": "description for this file",
              "downloadUrl": "/url-2",
              "name": "file number 2.jpg",
            },
          ]
        }
        itemCount={2}
      >
        <DataTable.Table />
      </DataTable>
    </Collapsible.Body>
  </Collapsible.Advanced>
</Card>
`;
